src/Global/organisms/MainMenu.tsx

```diff
@@ -127,7 +138,14 @@ const MainMenu: React.FC<IMenu> = ({ content, home = "/", logoMark, logoText, ke
   const toggleMenuPin = useCallback((e: any) => {
     if (e.pointerType === 'touch') { return; }
     togglePinned();
-  }, [togglePinned]);
+    if (showMenuOpen) {
+      setShowMenuOpen(false);
+      localStorage.setItem('isMenuOpen', 'false');
+    } else {
+      setShowMenuOpen(true);
+      localStorage.setItem('isMenuOpen', 'true');
+    }
```

Comment:
Can we move this local storage set and get methods inside the hook and use the `isMenuPinned` property here?

Reply:
Fixed